aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -114,6 +115,33 @@
    */
   public static final String S3FILEIO_ACL = "s3fileio.acl";
 
+  /**
+   * If {@link org.apache.iceberg.aws.glue.GlueCatalog} should use external lock or not, default to false.
+   * If set to true, it will use DynamoDB to enforce locking during commits.
+   */
+  public static final String GLUE_CATALOG_LOCK_ENABLED = "gluecatalog.lock.enabled";
+  public static final boolean GLUE_CATALOG_LOCK_ENABLED_DEFAULT = false;
```

Comment:
I think this should default to true because it is needed for correct operations.

Reply:
Yes I agree, but after a second thought, there are a few considerations for making this false by default:
1. this is an additional AWS resource that charges the user if enabled, so it is better to let people explicitly do that
2. for data warehouse ETLM use cases, the current Glue concurrent modification exception is good enough as a safe guard. I think it makes more sense for user to turn this on when they know there is a potential for conflict commits in use cases like streaming.

@yyanyy and @giovannifumarola, what do you think?

Reply:
I think it is important to always be correct. If we want users to need to opt in, then we can add a configuration setting that must be supplied, like the dynamo table. If that isn't supplied, then loading the catalog fails with an error message. Then we know that the user configured a specific dynamo table and understands that dynamo will be used.

Reply:
+1, also I'm not an expert in AWS free tier but according to [this](https://aws.amazon.com/free/?all-free-tier.sort-by=item.additionalFields.SortRank&all-free-tier.sort-order=asc&all-free-tier.q=dynamodb&all-free-tier.q_operator=AND) it seems like low provisioned capacity counts towards free tier, so charging might not be a big problem in normal use cases. If this is the case, I guess this also applies to how table gets set up (`billingMode`).